editors/sc-ide/widgets/lang_status_box.cpp

```diff
@@ -39,7 +41,27 @@ LangStatusBox::LangStatusBox(ScProcess* lang, QWidget* parent): StatusBox(parent
     connect(lang, SIGNAL(stateChanged(QProcess::ProcessState)), this,
             SLOT(onInterpreterStateChanged(QProcess::ProcessState)));
 
+    // initialize formats from settings:
+    auto const main = Main::instance();
+    applySettings(main->settings());
+    connect(main, &Main::applySettingsRequest, this, &LangStatusBox::applySettings);
+
     onInterpreterStateChanged(lang->state());
+
+    mLang = lang;
+}
+
+void LangStatusBox::applySettings(Settings::Manager* settings) {
+    backgroundColor = settings->getThemeVal("text").background().color();
+
+    notRunningColor = settings->getThemeVal("text").foreground().color();
+    runningColor = settings->getThemeVal("symbol").foreground().color();
+    startingColor = settings->getThemeVal("number").foreground().color();
```

Comment:
Similarly to other comments, I'd suggest "postwindowsuccess" and "postwindowwarning" colors here.